subsys/disk/disk_access.c

```diff
@@ -21,15 +21,15 @@ LOG_MODULE_REGISTER(disk);
 static sys_dlist_t disk_access_list = SYS_DLIST_STATIC_INIT(&disk_access_list);
 
 /* lock to protect storage layer registration */
-static K_MUTEX_DEFINE(mutex);
+static struct k_spinlock lock;
 
 struct disk_info *disk_access_get_di(const char *name)
 {
 	struct disk_info *disk = NULL, *itr;
 	size_t name_len = strlen(name);
 	sys_dnode_t *node;
 
-	k_mutex_lock(&mutex, K_FOREVER);
+	k_spinlock_key_t spinlock_key = k_spin_lock(&lock);
```

Comment:
k_spinlock_key_t definition should be at the top of the block, like other variables.